### PR TITLE
Feat: add the project api

### LINKED ISSUE #2895: [Feature] Implement list and create api for the project model.
**Is your feature request related to a problem? Please describe.**

In version 1.2, basic project management is implemented.

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

Fixes #2895

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

@wonderflow  @yangsoon

### CODE REVIEW

File: pkg/apiserver/model/application.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type Application struct {
 	Model
 	Name        string            `json:"name"`
 	Alias       string            `json:"alias"`
+	ProjectName string            `json:"projectName"`
```
Comment (wonderflow) [inline pkg/apiserver/model/application.go:35]:
```suggestion
	Project string            `json:"projec"`
```

File: pkg/apiserver/rest/apis/v1/types.go [hunk 2]
```diff
@@ -253,9 +253,9 @@ type ClusterBase struct {
 
 // ListApplicatioOptions list application  query options
 type ListApplicatioOptions struct {
-	Namespace  string `json:"namespace"`
-	TargetName string `json:"targetName"`
-	Query      string `json:"query"`
+	ProjectName string `json:"projectName"`
```
Comment (wonderflow) [inline pkg/apiserver/rest/apis/v1/types.go:256]:
```suggestion
	Project string `json:"project"`
```

File: pkg/apiserver/rest/apis/v1/types.go [hunk 3]
```diff
@@ -306,7 +306,7 @@ type ApplicationStatisticsResponse struct {
 type CreateApplicationRequest struct {
 	Name        string                  `json:"name" validate:"checkname"`
 	Alias       string                  `json:"alias" validate:"checkalias" optional:"true"`
-	Namespace   string                  `json:"namespace" validate:"checkname"`
+	ProjectName string                  `json:"projectName" validate:"checkname"`
```
Comment (wonderflow) [inline pkg/apiserver/rest/apis/v1/types.go:309]: 同上，就叫 project

File: pkg/apiserver/rest/usecase/project.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+		Alias:       req.Alias,
+		Namespace:   fmt.Sprintf("project-%s", req.Name),
+	}
+
+	// create namespace at first
+	if err := p.kubeClient.Create(ctx, &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: new.Namespace,
+			Labels: map[string]string{
+				oam.LabelProjectNamesapce: new.Name,
```
Comment (wonderflow) [inline pkg/apiserver/rest/usecase/project.go:117]: 这一行跟 req.Name 一样吧？ 感觉应该加的是一个固定的label，而不是这种变化的。

类似这种：
`velaux.oam.dev:project`

Comment (wonderflow) [reply pkg/apiserver/rest/usecase/project.go:117]: 最终你可能要通过这种label反查哪些namespace是直接显示的

Comment (barnettZQG) [reply pkg/apiserver/rest/usecase/project.go:117]: 这里不需要反查，加标签的目的是准确标记其是属于某个project的。

**Review by wonderflow — APPROVED** [review]

**Review by yangsoon — APPROVED** [review]

**Pull request merged**